### ISSUE TITLE
cargo build issue on Apple Silicon M1 Mac

### ISSUE THREAD

**Issue by reportaman** [chipsenkbeil/distant#54]
``>> cargo build --release``

```
...
   Compiling futures v0.3.17
   Compiling tokio-util v0.6.8
   Compiling serde_cbor v0.11.2
   Compiling distant-core v0.14.2 (/Users/reportaman/Downloads/distant/core)
error[E0658]: use of unstable library feature 'str_split_once': newly added
   --> core/src/client/lsp/data.rs:280:47
    |
280 |             if let Some((name, value)) = line.split_once(':') {
    |                                               ^^^^^^^^^^
    |
    = note: see issue #74773 <https://github.com/rust-lang/rust/issues/74773> for more information

error[E0599]: no variant or associated item named `Unsupported` found for enum `std::io::ErrorKind` in the current scope
   --> core/src/data.rs:632:28
    |
632 |             io::ErrorKind::Unsupported => Self::Unsupported,
    |                            ^^^^^^^^^^^ variant or associated item not found in `std::io::ErrorKind`

error[E0599]: no variant or associated item named `OutOfMemory` found for enum `std::io::ErrorKind` in the current scope
   --> core/src/data.rs:633:28
    |
633 |             io::ErrorKind::OutOfMemory => Self::OutOfMemory,
    |                            ^^^^^^^^^^^ variant or associated item not found in `std::io::ErrorKind`

error[E0658]: use of unstable library feature 'str_split_once': newly added
  --> core/src/server/port.rs:67:17
   |
67 |         match s.split_once(':') {
   |                 ^^^^^^^^^^
   |
   = note: see issue #74773 <https://github.com/rust-lang/rust/issues/74773> for more information

error: aborting due to 4 previous errors

Some errors have detailed explanations: E0599, E0658.
For more information about an error, try `rustc --explain E0599`.
error: could not compile `distant-core`

To learn more, run the command again with --verbose.

~/Downloads/distant master 19s
```

```
❯ neofetch
                    'c.          reportaman@Amans-Mac-mini.local
                 ,xNMM.          ------------------------------
               .OMMMMo           OS: macOS 11.6 20G165 arm64
               OMMM0,            Host: Macmini9,1
     .;loddo:' loolloddol;.      Kernel: 20.6.0
   cKMMMMMMMMMMNWMMMMMMMMMM0:    Uptime: 4 hours, 1 min
 .KMMMMMMMMMMMMMMMMMMMMMMMWd.    Packages: 1 (brew)
 XMMMMMMMMMMMMMMMMMMMMMMMX.      Shell: zsh 5.8
;MMMMMMMMMMMMMMMMMMMMMMMM:       Resolution: 3840x2160
:MMMMMMMMMMMMMMMMMMMMMMMM:       DE: Aqua
.MMMMMMMMMMMMMMMMMMMMMMMMX.      WM: Quartz Compositor
 kMMMMMMMMMMMMMMMMMMMMMMMMWd.    WM Theme: Blue (Dark)
 .XMMMMMMMMMMMMMMMMMMMMMMMMMMk   Terminal: tmux
  .XMMMMMMMMMMMMMMMMMMMMMMMMK.   CPU: Apple M1
    kMMMMMMMMMMMMMMMMMMMMMMd     GPU: Apple M1
     ;KMMMMMMMWXXWMMMMMMMk.      Memory: 2003MiB / 16384MiB
       .cooc,.    .,coo:.






```

**Comment by chipsenkbeil** [comment]
You have too old of a version of Rust. `str_split_once` was introduced in 1.52.0.
Unsupported in 1.53.0. OutOfMemory in 1.54.0.

So either give 1.54.0 a try or you can download the latest build from the releases section :)

**Comment by chipsenkbeil** [comment]
@reportaman as of https://github.com/chipsenkbeil/distant/commit/32639166bca1de367be91435d38a3eca376621aa, I've refactored parts of distant to now work with a MSRV of 1.51.0, which I believe will also fix your build challenges.

This is now tested as part of the automated CI and highlighted in both distant and distant-core READMEs, so thanks for bringing this to my attention. :)

Closing this one out.
